src/include/rados.h

```diff
@@ -562,6 +563,18 @@ struct ceph_osd_op {
 	__le32 payload_len;
 } __attribute__ ((packed));
 
+/*
+ * Check the compatibility of struct ceph_osd_op
+ *  (2+4+(2*8+8+4)+4) = (sizeof(ceph_osd_op::op) +
+ *                     sizeof(ceph_osd_op::flags) +
+ *                     sizeof(ceph_osd_op::extent) +
+ *                     sizeof(ceph_osd_op::payload_len))
+ */
+#ifdef __cplusplus
+static_assert(sizeof(ceph_osd_op) == (2+4+(2*8+8+4)+4),
```

Comment:
> Also, this static_assert involves some magic number "2+4+(2*8+8+4)+4". Do I need to add a global constant value to represent this magic number, i.e. "const uint32_t MAX_BYTES_OF_CEPH_OSD_OP = 38;"

ahh, i just realized that `rados.h` is shared by C and C++ source files! `MAX_BYTES_OF_CEPH_OSD_OP` does not seem right. the size of `ceph_osd_op` **must** be exactly 38 bytes. and we can only add more fields at the end of union structs (`extent` excluded), and the new behavior should be backward compatible if the newly added field(s) reads zero. this static assert is just a better-than-nothing message to posterity.

but i think the comment is good enough to serve this purpose. what do you think?